go/vt/vtgate/planbuilder/route.go

```diff
@@ -519,10 +520,13 @@ func (rb *route) Wireup(bldr builder, jt *jointab) error {
 				return
 			}
 		case sqlparser.TableName:
-			if node.Qualifier != infoSchema {
+			if !systemTable(node.Qualifier.String()) {
 				node.Name.Format(buf)
 				return
 			}
+			// We accept INFORMATION_SCHEMA, but underlying mysql might not
```

Comment:
i personally think we should let the underlying mysql do throw an error.  the idea here is vitess should be invisible. but if it doesnt work underlying then it doesn't work. 

Separately, I also think the less we rewrite queries the better, but that may be a lost cause. It's a gotcha when grepping logs and such for patterns you expect.

Reply:
i'll let @giaquinti weigh in

Reply:
I'm +1 to don't rewrite the query